### PR TITLE
moved dataset_cache to function 

### PR DESCRIPTION
**Pull request by crewtool**
# Description
Passing folder for cache was badly handled. I moved getting path to huggingface datasets funciont 

# Checklist
Answer each question with "yes", and provide an explanation if the answer isn't what is expected.

1. Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - Answer: yes
2. Are all variables set with logical defaults that are backward compatible?
   - Answer: yes
3. Can you provide a link to a Neptune experiment verifying the functionality of this PR?
   - Answer: na
4. Have you successfully executed the linter and tests?
   - Answer: yes
5. Are all functions concise and don't require splitting?
   - Answer: yes 
6. Is there documentation for arguments and complex logic?
   - Answer: yes 
7. Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - Answer: no
8. Are all functions placed in appropriate files?
   - Answer: yes 
9. Is the PR name meaningful and descriptive?
   - Answer: yes 
10. Is PR description provided, or unnecessary?
    - Answer: yes
  
Other questions:

11. Have you not made changes to the requirements or anything related to the Singularity image? If you have, did you generate and upload a new image to the clusters?
    - Answer: no
12. Is the change significant enough to notify all individuals working with this repository?
    - Answer: yes
13. Is there a need for a second reviewer?
    - Answer: no

### CODE REVIEW

**Review by kuba-krj — APPROVED** [review]
LGTM. Tylko proszę zmień ścieżkę na entropy gpu na naszą nowa przestrzeń `/common/llm-random`

File: lizrd/scripts/grid.py [hunk 1]
```diff
@@ -52,6 +53,8 @@
 COPIED_CODE_PATH = ""
 if __name__ == "__main__":
     runner = get_machine_backend()
+    CACHE_PATH = os.getenv("CACHE_PATH", get_cache_path(runner))
+    HF_DATASETS_CACHE = os.getenv("HF_DATASETS_CACHE", CACHE_PATH)
```
Comment (maciek-pioro) [inline lizrd/scripts/grid.py:57]: Why are we splitting this into two different things, i.e. CACHE_PATH and HF_DATASETS_CACHE?

File: lizrd/scripts/grid_utils.py [hunk 2]
```diff
@@ -27,6 +27,17 @@ def get_machine_backend() -> MachineBackend:
...
 
+def get_cache_path(runner: MachineBackend) -> str:
+    if runner == MachineBackend.ATHENA:
+        return "/net/tscratch/people/plgjkrajewski/.cache"
+    elif runner == MachineBackend.IDEAS:
+        return "/raid/NFS_SHARE/llm-random/.cache"
+    elif runner == MachineBackend.ENTROPY_GPU:
+        return "/home/jkrajewski/.cache"
+    else:
+        return "/tmp/.cache"
```
Comment (maciek-pioro) [inline lizrd/scripts/grid_utils.py:38]: Doesn't this lead to cache disappearing on every reboot and datasets having to be downloaded again?

**Review by maciek-pioro — COMMENTED** [review]
Two comments, maybe changes should be requested.

File: lizrd/scripts/grid.py (hunk 1 above)
Comment (crewtool) [reply lizrd/scripts/grid.py:58]: Make sense

File: lizrd/scripts/grid_utils.py [hunk 3]
```diff
@@ -27,6 +27,28 @@ def get_machine_backend() -> MachineBackend:
...
 
+def get_common_directory(runner: MachineBackend) -> str:
+    if runner == MachineBackend.ATHENA:
+        return "/net/pr2/projects/plgrid/plggllmeffi"
+    elif runner == MachineBackend.IDEAS:
+        return "/raid/NFS_SHARE/llm-random"
+    elif runner == MachineBackend.ENTROPY_GPU:
+        return "/common/llm-random"
+    else:
+        return "~"
```
Comment (maciek-pioro) [inline lizrd/scripts/grid_utils.py:38]: Preferably os.getenv("HOME") for a couple of reasons xd

File: lizrd/scripts/grid_utils.py [hunk 4]
```diff
@@ -27,6 +27,28 @@ def get_machine_backend() -> MachineBackend:
...
+
+
+def get_cache_path(runner: MachineBackend) -> str:
+    common_dir = get_common_directory(runner)
+    return f"{common_dir}/.cache"
```
Comment (maciek-pioro) [inline lizrd/scripts/grid_utils.py:43]: Default HF folder would be ~/.cache/huggingface/datasets (at least locally). Maybe let's somehow keep that?

File: lizrd/scripts/grid_utils.py (hunk 3 above)
Comment (crewtool) [reply lizrd/scripts/grid_utils.py:38]: fair point

File: lizrd/scripts/grid_utils.py (hunk 4 above)
Comment (crewtool) [reply lizrd/scripts/grid_utils.py:43]: Maybe f"{os.getenv('HOME')}/.cache/huggingface/datasets"?

**Review by maciek-pioro — APPROVED** [review]
Well done 💯

**Pull request merged**